locale/inc/leatherman/locale/locale.hpp

```diff
@@ -95,63 +95,161 @@ namespace leatherman { namespace locale {
      * @param domain The catalog domain to use for i18n via gettext.
      * @return The translated string.
      */
-    std::string translate_c(std::string const& context, std::string const& single, std::string const& plural, int n, std::string const& domain = PROJECT_NAME);
+    std::string translate_np(std::string const& context, std::string const& single, std::string const& plural, int n, std::string const& domain = PROJECT_NAME);
 
-#ifdef LEATHERMAN_I18N
     /**
-     * A formatter class for translating and formatting text according to a domain.
+     * Translates and formats text using the locale initialized by this library.
+     * @param trans The translation function.
+     * @param args Format arguments.
+     * @return The string generated by translating the format string, then applying the arguments.
      */
-    class formatter
+    template <typename... TArgs>
+    std::string _format_common(std::function<std::string(const std::string&)>&& trans, TArgs... args)
```

Comment:
One technique to look up here is anonymous namespaces. Essentially makes the function private to this namespace.